tests/test_other.py

```diff
@@ -1443,7 +1465,7 @@ def test(link_flags, lib_suffix):
       # Build libfile normally into an .so
       self.run_process([EMCC, os.path.join('libdir', 'libfile.cpp'), '-shared', '-o', os.path.join('libdir', 'libfile.so' + lib_suffix)])
       # Build libother and dynamically link it to libfile
-      self.run_process([EMCC, os.path.join('libdir', 'libother.cpp')] + link_flags + ['-shared', '-o', os.path.join('libdir', 'libother.so')])
+      self.run_process([EMCC, '-Llibdir', os.path.join('libdir', 'libother.cpp')] + link_flags + ['-shared', '-o', os.path.join('libdir', 'libother.so')])
```

Comment:
@sbc100 this fixes the failing test. Can you check if this is indeed a legitimate fix for a previously uncovered oversight, or just masking regression?

Lines from old testing result (same for test-other and test-mac):

```
wasm-ld: error: unable to find library -lfile
emcc: error: '/root/emsdk/upstream/bin/wasm-ld -o libdir/libother.so /tmp/tmph4b2_1xa/emscripten_temp_eidmkp_z/libother_0.o -lfile -L/root/emsdk/upstream/emscripten/cache/sysroot/lib/wasm32-emscripten --relocatable -mllvm -combiner-global-alias-analysis=false -mllvm -enable-emscripten-sjlj -mllvm -disable-lsr' failed (1)
test_multidynamic_link (test_other.other) ... FAIL
```

Reply:
As for compatibility issues:

1. If user ported the build from another platform, then build system would have already caught such issues on that platform.
2. This unmasks incorrect build options that used to be ignored otherwise.
3. As you've told me, '-r' is rare nowadays.

Reply:
It looks correct to me... but I also have no idea why this worked before your change.

Reply:
@sbc100 It just copied it around while dropping the linker flags.